src/java/main/org/apache/zookeeper/server/quorum/Learner.java

```diff
@@ -390,6 +391,7 @@ else if (qp.getType() == Leader.SNAP) {
                             + Long.toHexString(qp.getZxid()));
                     System.exit(13);
                 }
+                zk.getZKDatabase().setlastProcessedZxid(qp.getZxid());
```

Comment:
@hanm, Do we have any issue created to handle this case, if not please create one for tracking this change. Thanks!